app/src/lib/components/PullRequestCard.svelte

```diff
@@ -115,67 +120,123 @@
-		if (!status.hasChecks) return;
-		if (status.error) return 'error-small';
-		if (status.completed) {
-			return status.success ? 'success-small' : 'error-small';
+	): {
+		color: TagStyle;
+		icon: keyof typeof iconsJson;
+		text: string;
+	} {
+		if (checksError || detailsError) {
+			return { color: 'error', icon: 'warning-small', text: 'Failed to load' };
 		}
 
-		return 'spinner';
-	}
+		if (fetching || !status) {
+			return { color: 'neutral', icon: 'spinner', text: 'Checks' };
+		}
 
-	function getChecksText(status: ChecksStatus | undefined | null): string | undefined {
-		if (checksError || detailsError) return 'Failed to load';
-		if (!status) return 'Checks';
-		if (!status.hasChecks) return 'No checks';
-		if (status.error) return 'error';
 		if (status.completed) {
-			return status.success ? 'Checks passed' : 'Checks failed';
+			const color = status.success ? 'success' : 'error';
+			const icon = status.success ? 'success-small' : 'error-small';
+			const text = status.success ? 'Checks passed' : 'Checks failed';
+			return { color, icon, text };
 		}
-		// Checking this second to last let's us keep the previous tag color unless
-		// checks are currently running.
-		if (isFetchingChecks) return 'Checks';
-		return 'Checks are running';
-	}
 
-	function getPrStatusLabel(pr: DetailedPullRequest | undefined): string {
-		if (pr?.mergedAt) return 'Merged';
-		if (pr?.closedAt) return 'Closed';
-		if (pr?.draft) return 'Draft';
-		return 'Open';
+		return {
+			color: 'warning',
+			icon: 'spinner',
+			text: getChecksCount(status)
+		};
 	}
 
-	function getStatusIcon(pr: DetailedPullRequest | undefined): keyof typeof iconsJson | undefined {
-		if (pr?.mergedAt) return 'merged-pr-small';
-		if (pr?.closedAt) return 'closed-pr-small';
-		if (pr?.closedAt) return 'draft-pr-small';
-		return 'pr-small';
+	function getPrStatusInfo(pr: DetailedPullRequest | undefined): {
+		label: string;
+		icon: keyof typeof iconsJson | undefined;
+		color: TagStyle;
+	} {
+		if (!pr) {
+			return { label: 'Status', icon: 'spinner', color: 'neutral' };
+		}
+
+		if (pr?.mergedAt) {
+			return { label: 'Merged', icon: 'merged-pr-small', color: 'purple' };
+		}
+
+		if (pr?.closedAt) {
+			return { label: 'Closed', icon: 'closed-pr-small', color: 'error' };
+		}
+
+		if (pr?.draft) {
+			return { label: 'Draft', icon: 'draft-pr-small', color: 'neutral' };
+		}
+
+		return { label: 'Open', icon: 'pr-small', color: 'success' };
 	}
 
-	function getStatusColor(pr: DetailedPullRequest | undefined): TagStyle {
-		if (pr?.mergedAt) return 'purple';
-		if (pr?.closedAt) return 'error';
-		if (pr?.draft) return 'neutral';
-		return 'success';
+	function getInfoMessageInfo(
+		pr: DetailedPullRequest | undefined,
+		mergeableState: string | undefined,
+		checksStatus: ChecksStatus | null | undefined,
+		isFetchingChecks: boolean
+	):
+		| {
+				icon: keyof typeof iconsJson;
+				style: MessageStyle;
+				text: string;
+		  }
+		| undefined {
```

Comment:
How about defining a type for this? Looks like there's repetitions, but one of them has `label` instead of `text`.